packages/merkle-distributor/src/MerkleDistributorHelper.ts

```diff
@@ -0,0 +1,83 @@
+// This helper includes a set of useful functions when dealing with merkle trees and payout claims.
+const { ethers } = require("ethers");
+const { getAbi } = require("@uma/core");
+
+import MerkleTree from "./MerkleTree";
+const CloudflareKVHelper = require("../src/CloudflareKVHelper");
+
+// keccak256(abi.encode(account, amount))
+const createLeaf = (account: string, amount: string) => {
+  return Buffer.from(ethers.utils.solidityKeccak256(["address", "uint256"], [account, amount]).substr(2), "hex");
+};
+
+function createMerkleDistributionProofs(
+  recipientsData: { [key: string]: { amount: string; metaData: any } },
+  windowIndex: number
+) {
+  // Build an array of leafs for each recipient This is simply a hash of the address and recipient amount.
+  const recipientLeafs = Object.keys(recipientsData).map((recipientAddress: string) =>
+    createLeaf(recipientAddress, recipientsData[recipientAddress].amount)
+  );
+
+  // Build the merkle proof from the leafs.
+  const merkleTree = new MerkleTree(recipientLeafs);
+
+  // Lastly, append the leaf and generated proof for each recipient to the original data structure.
+  const recipientsDataWithProof: any = {};
+  Object.keys(recipientsData).forEach((recipientAddress, index) => {
+    recipientsDataWithProof[recipientAddress] = {};
+    recipientsDataWithProof[recipientAddress].amount = recipientsData[recipientAddress].amount;
+    recipientsDataWithProof[recipientAddress].metaData = recipientsData[recipientAddress].metaData;
+    recipientsDataWithProof[recipientAddress].windowIndex = windowIndex;
+    recipientsDataWithProof[recipientAddress].proof = merkleTree.getHexProof(recipientLeafs[index]);
+  });
+
+  return { recipientsDataWithProof, merkleRoot: merkleTree.getHexRoot() };
+}
+
+async function getClaimsForAddress(merkleDistributorAddress: string, claimerAddress: string, chainId: number) {
+  // Create a new ethers contract instance to fetch on-chain contract information.
+  const infuraApiKey = process.env.INFURA_API_KEY || null;
+  const ethersProvider = new ethers.providers.InfuraProvider(chainId, infuraApiKey);
+  const signer = new ethers.VoidSigner(claimerAddress, ethersProvider);
+  const merkleDistributorContract = new ethers.Contract(merkleDistributorAddress, getAbi("MerkleDistributor"), signer);
```

Comment:
can we pass in the provider as a dep to this function? also im not sure you need this void signer to call the contract, you can just [pass in the provider](https://docs.ethers.io/v5/api/contract/example/#example-erc-20-contract--connecting-to-a-contract--erc20contract)

Reply:
Yup! you _could_. The reason I did it this way is that I wanted the _caller_ of this function to not have to have a provider as it will most likely be called by a serverless function, that the front end would call. It also decouples the method from the provider and makes it provider agnostic.

Reply:
i see what you are saying. yea it does make it convenient for a serverless call, but the logic just feels out of place among the rest of the functions in this file.  my last suggestion would be to create serverless folder which has thin js wrapper for configuring the dependencies and making a call to the low level function in this file.  this folder could also store any other future serverless calls. Again this is just more about design, separating config from business logic, but functionally it will work the same.